include/boundary_region.hxx

```diff
@@ -18,11 +20,14 @@ enum BndryLoc {BNDRY_XIN=1,
 
 class BoundaryRegionBase {
 public:
-  BoundaryRegionBase() {}
-  BoundaryRegionBase(const string &name) : label(name) {}
-  BoundaryRegionBase(const string &name, BndryLoc loc) : label(name), location(loc) {}
+  BoundaryRegionBase() = delete;
+  BoundaryRegionBase(Mesh* mesh) : localmesh(mesh) {}
```

Comment:
This changes the interface! An alternative would be:
```cpp
BoundaryRegionBase(Mesh* localmesh = nullptr) : localmesh(localmesh ? localmesh : mesh) {}
```
to use the global `mesh` if one isn't provided.